### ISSUE TITLE
Evaluate "tuplet loss"

### ISSUE THREAD

**Issue by hominot** [hominot/research#17]
Hyperparameters:
- Parametrization (Euclidean, dot product)
- L2 regularizing parameter
- (N+1)-pair size (N=2, 71, 191 ...)
- Learning rate
